core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +158,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      condOpt(server.getThreadPool) { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
I had to look up `condOpt` here .. is this just trying to operate on it only if non-null and a `LifeCycle`? I personally think it's more idiomatic to:

`Option(server.getThreadPool).collect { case l : LifeCycle => l }.foreach(_.stop())`